core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -488,6 +494,11 @@ public static long mapRW(FilesFacade ff, long fd, long size, int memoryTag) {
         return mapRW(ff, fd, size, 0, memoryTag);
     }
 
+    /**
+     * Maps a file in read-write mode.
+     *
+     * Important note. Linux requires the offset to be page aligned.
+     */
     public static long mapRW(FilesFacade ff, long fd, long size, long offset, int memoryTag) {
         allocateDiskSpace(ff, fd, size + offset);
         long addr = ff.mmap(fd, size, offset, Files.MAP_RW, memoryTag);
```

Comment:
Same assert can go here too

Reply:
Added assert in 9678f2dec1966a917f63dd78de06e01c1d2d0740